### DISCUSSION TITLE
How to add top level folder for C/C++ builds in eclipse

### DISCUSSION THREAD

**Discussion by ngoliNI** [eclipse-cdt/cdt#301] [General]
When I add a folder in 'Includes', the .h files under its subfolders are not getting picked up.
But when added the exact file path for a header file, it works.

Can anyone suggest on how to add the top-level directory so that the build path searches its respective subfolders also?

**Comment by jonahgraham** [comment]
Not sure I fully follow the question. A screenshot of your setting may help clarify. 

Here is what I would setup to include files in my project:

![image](https://user-images.githubusercontent.com/679236/221616822-7d6791a6-3f86-4c36-a605-136bcc1e58a4.png)

The bit of magic is making your include directory in the setting something like `"${workspace_loc:/${ProjName}/src/myincludes}"`

**Reply by ngoliNI** [reply]
<img width="395" alt="Screenshot 2023-02-27 215803" src="https://user-images.githubusercontent.com/126497477/221621960-bdec6986-31b2-45c9-a4ac-ccf8dce1edd0.png">
Thank you for the response.
I am calling some MATLAB header files in a program. And that header file has multiple dependencies. 
Adding each of the header file paths in the 'Includes' is tedious as there are multiple paths. Is there any way, if I add the top-level directory, the paths from their subdirectories get picked up without manually adding all of them as all of the header paths are in different sub-directories?

**Reply by jonahgraham** [reply]
_Thanks for clarifying._

No, there is no way I know of to do that. 

Using some shell and makefile tricks could potentially enable such a flow, but as you are using the Eclipse CDT internal builder, even those aren't an option.
